### ISSUE TITLE
If the accesskey and token expire some time after connecting to signaling server,can our device be able to send or receive signaling under an established connection? 

### ISSUE THREAD

**Issue by arthur-zou** [awslabs/amazon-kinesis-video-streams-webrtc-sdk-c#272]

**Comment by MushMal** [comment]
@arthur-zou the short answer is NO. 

TL;DR The signaling client uses the supplied credential provider to create the signaling client (making front-end calls) and then make the initial connection. The signaling client will keep the connection until the backend drops the connection on max connection duration or if there is an error reported. In either case, the signaling client (if configured so which is the default) will attempt to re-connect. The signaling client state machinery will attempt to get the fresh credentials again from the credentials provider and will use those. If the credentials are stale/bad, it will eventually fail and will call the onError callback.

I am resolving this issue. Please feel free to re-open if you need more info.
